### PR TITLE
[IMP] Cache system for RPC

### PR DESCRIPTION
**Pull request by mdo-odoo**

### CODE REVIEW

File: addons/request_cache/request_cache.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        for model in self.pool.itervalues():
+            cache_dependencies = getattr(model, '_cache_dependencies', False)
+            if cache_dependencies:
+                models      = set()
+                core_model  = model._name
+                models.add(core_model)
+                inherits    = model._inherits
+                if inherits:
+                    for m in inherits:
+                        models.add(m)
```
Comment (rco-odoo) [inline addons/request_cache/request_cache.py]: you can have several levels of `_inherits` -> you need a recursive traversal of parent models

File: addons/request_cache/request_cache.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+                for m in cache_dependencies:
+                    models.add(m)
+
+                for model in models:
+                    model = model.replace('.', '_')
+                    trigger = "DROP TRIGGER IF EXISTS after_change_%s on %s; CREATE TRIGGER after_change_%s AFTER INSERT OR UPDATE OR DELETE ON %s EXECUTE PROCEDURE invalid_cache('%s');" % ((model,)*4 + (core_model,))
+                    cr.execute(trigger)
```
Comment (rco-odoo) [inline addons/request_cache/request_cache.py]: The trigger query is unreadable, split it on several lines and use format expressions like `"%(name)s"`.

Comment (rco-odoo) [inline addons/request_cache/request_cache.py]: Bug: `model.replace('.', '_')` is the default name for a model's table.
Use the attribute `_table` on the model instead.

File: addons/request_cache/request_cache.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+class request_cache_store(models.Model):
+    _name = 'cache.store'
+
+    hashkey = fields.Char(string='Cache Group Name', required=True, readonly=True)
+    model   = fields.Char(string='Cached Model Name', required=True, readonly=True)
+    data    = fields.Text(string='Serialized Data', required=True, readonly=True) 
+    
+    def init(self, cr):
+        cr.execute("CREATE OR REPLACE FUNCTION invalid_cache() RETURNS TRIGGER AS ' BEGIN DELETE FROM cache_store WHERE model=TG_ARGV[0]; RETURN NEW; END; ' LANGUAGE 'plpgsql';");
+
```
Comment (rco-odoo) [inline addons/request_cache/request_cache.py]: split your query on several lines, please

**Comment by antonylesuisse** [comment]
Cancelled too intrusive we will merge a pos-only specific version.
But this might be usefull for somebody else.

**Pull request closed without merging**